test/graphql-api/multipart.test.ts

```diff
@@ -0,0 +1,60 @@
+import * as path from 'path'
+import { runBrowserWith } from '../support/runBrowserWith'
+import { executeOperation } from './utils/executeOperation'
+
+function createRuntime() {
+  return runBrowserWith(path.resolve(__dirname, 'multipart.mocks.ts'))
+}
+
+test('accepts a file from a GraphQL mutation', async () => {
+  const runtime = await createRuntime()
+  const UPLOAD_MUTATION = `
+    mutation UploadFile(
+      $file1: Upload
+      $file2: Upload
+      $plainText: String
+      ) {
+      multipart(
+        file1: $file1
+        file2: $file2
+        plainText: $plainText
+        ){
+        file1
+        file2
+        plainText
+      }
+    }
+  `
+  const res = await executeOperation(
+    runtime.page,
+    {
+      query: UPLOAD_MUTATION,
+      variables: {
+        file1: null,
+        file2: null,
+        files: [null, null],
+        plainText: 'text',
+      },
+    },
+    {
+      map: {
+        '0': ['variables.file1', 'variables.files.0'],
```

Comment:
Can this be an Array instead?

```js
map: [
  ['file1', 'file2'],
  [c, d]
]
```

Reply:
As far as I read [the spec](https://github.com/jaydenseric/graphql-multipart-request-spec#examples) , `map` should be an object, not an array.

https://github.com/graphql-python/gql/blob/c17959c453c4f9c631c31ddb987ebbd3202b971a/gql/transport/aiohttp.py#L167-L169

This is python implementation of GraphQL client. It builds `map` as `dict` type that will be encoded to object in JSON.

Reply:
Got it 👍 No harm keeping it this way, just made me curious.